lib/library.hoon

```diff
@@ -0,0 +1,135 @@
+/-  *post, resource, library, spider
+/+  *graph-store, strandio
+|%
+++  incr-index
+  :: increments the last value in the index by 1
+  |=  [=index:post]
+  ^-  index:post
+  =/  last=@                (sub (lent index) 1)  :: calculate the array index of the last item
+  =/  old-revision-count=@  (snag last index)     :: get the last item
+  (snap index last (add 1 old-revision-count))    :: replace the value of last item with 1 added to it
+++  scry-for
+  |*  [=mold =path]
+  .^  mold
+    %gx
+    (scot %p our.bowl)
+    %library-proxy
+    (scot %da now.bowl)
+    (snoc `^path`path %noun)
+  ==
+::
+++  make-meta-contents
+  |=  [=book:library]
+  ^-  (list content)
+  ~[[%text (crip title.book)] [%text (crip isbn.book)]]
+++  make-comment-contents
+  |=  [comment-text=comment:library]
+  ^-  (list content)
+  ~[[%text (crip comment-text)]]
+::
+++  is-allowed
+  |=  [requester=ship host=ship =policy:library]
+  ^-  ?
+  ?:  =(requester host)  :: host is always allowed
+    %.y
+  ?-  -.policy
+      %open       %.y
+      %children   (team:title host requester)
+      %whitelist  (~(has in ships.policy) requester)
+  ==
+++  index-to-path
+  |=  [idx=index:post]
+  ^-  path
+  (turn idx (cury scot %ud))
+::
+::
+++  create-library-update
+  |=  [[owner=ship name=@tas] time-sent=time]
+  ^-  update
+  :-  time-sent
+  :-  %add-graph
+  :*  [owner name]
+      *graph  :: the official thread uses (gas:orm ~ ~), which is equivalent
+      [~ %graph-validator-library]
+      %.n     :: don't overwrite an existing library
+  ==
+++  remove-library-update
+  |=  [[owner=ship name=@tas] time-sent=time]
+  ^-  update
+  :-  time-sent
+  :-  %remove-graph
+  [owner name]
+++  add-book-update
```

Comment:
implement revisions data structure for book metadata

Reply:
this applies to `revise-book-update`, not this arm

Reply:
https://github.com/ynx0/library/pull/3/commits/54b2fc3c9317f13687c0fc8e3d93945a020a4dde